DH-14 Initiative

Dyrk84 opened the pull request:
Initiative variables added to characters with setters, getters.
The killedMonsterCounter moved in MonsterFactory.
New variables in Fight class.
New attackInitiating method in Fight class. Fight system reworked.
Text separator lines for easier readability.

Dyrk84 committed:
DH-14 Initiative
Initiative variables added to characters with setters, getters.
The killedMonsterCounter moved in MonsterFactory.
New variables in Fight class.
New attackInitiating method in Fight class. Fight system reworked.
Text separator lines for easier readability.

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/monsters/Goblin.java:
```diff
@@ -66,4 +72,14 @@ private void startValues() {
         return Dice.rollDice(maxDamage, numOfDices);
     }
+
+    @Override
+    public int getMonsterInitiative(){
+        return getInitiative();
+    }
+
+    @Override
+    public void setMonsterInitiative(int initiative){
+        this.initiative = initiative;
+    }
```
This is not needed if you're using @Getter and @Setter on the variable

Dyrk84 replied:
I deleted this lines, thanks!

Dyrk84 committed:
DH-14 Initiative
reworked attackInitiating method
(TODO tests, text separate lines)

Dyrk84 committed:
DH-14 Initiative
reworked attackInitiating method
new Tool: TextSeparator class
The tests are work

Dyrk84 committed:
reworked text separate lines

Dyrk84 committed:
reworked text separate lines
